From 8b5b8384fe717a8f41d2292b3c7b874bf2a9d2e5 Mon Sep 17 00:00:00 2001
From: xubing <bing.xu@intel.com>
Date: Tue, 26 Sep 2023 16:12:42 +0800
Subject: [PATCH] [WA]Fix JavaCrash on com.moji.mjweather.

Some views are used after they are removed on dispatchGenericPointerEvent,
we reset view array list size to avoid these views are reused again.

Tracked-On: OAM-111797
Signed-off-by: xubing <bing.xu@intel.com>
---
 core/java/android/view/ViewGroup.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/core/java/android/view/ViewGroup.java b/core/java/android/view/ViewGroup.java
index 679da313edcc..9ac484fa31d4 100644
--- a/core/java/android/view/ViewGroup.java
+++ b/core/java/android/view/ViewGroup.java
@@ -2541,6 +2541,10 @@ public abstract class ViewGroup extends View implements ViewParent, ViewManager
                     && isChildrenDrawingOrderEnabled();
             final View[] children = mChildren;
             for (int i = childrenCount - 1; i >= 0; i--) {
+                if (i > mChildrenCount -1) {
+                    i = mChildrenCount -1;
+                    Log.w(TAG, "reset array list size.....");
+                }
                 final int childIndex = getAndVerifyPreorderedIndex(childrenCount, i, customOrder);
                 final View child = getAndVerifyPreorderedView(preorderedList, children, childIndex);
                 if (!child.canReceivePointerEvents()
-- 
2.34.1

